build.gradle

```diff
@@ -9,10 +10,20 @@ repositories {
 
 dependencies {
     testImplementation 'junit:junit:4.12'
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
+}
+
+compileKotlin {
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+compileTestKotlin {
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
 }
 
 run {
     standardInput = System.in
 }
-
-mainClassName = 'RockPaperSmash.Driver'
```

Comment:
mainClassName has to be specified, but kotlin classes are named a little different. You'll probably need:

```groovy
 mainClassName = 'RockPaperSmash.DriverKt'
```

Reply:
These are the changes you need to make it work as expected with the application plugin:

```diff
diff --git a/.idea/misc.xml b/.idea/misc.xml
index bc8d0a3..29af3ee 100644
--- a/.idea/misc.xml
+++ b/.idea/misc.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="ExternalStorageConfigurationManager" enabled="true" />
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_11" project-jdk-name="11" project-jdk-type="JavaSDK">
     <output url="file://$PROJECT_DIR$/out" />
   </component>
 </project>
\ No newline at end of file
diff --git a/build.gradle b/build.gradle
index 2e01c8a..8eac665 100644
--- a/build.gradle
+++ b/build.gradle
@@ -27,3 +27,5 @@ compileTestKotlin {
 run {
     standardInput = System.in
 }
+
+mainClassName = 'RockPaperSmash.DriverKt'
diff --git a/src/main/kotlin/RockPaperSmash/CPU.kt b/src/main/kotlin/RockPaperSmash/CPU.kt
index fb3bec2..403e1d8 100644
--- a/src/main/kotlin/RockPaperSmash/CPU.kt
+++ b/src/main/kotlin/RockPaperSmash/CPU.kt
@@ -1,7 +1,5 @@
 package RockPaperSmash
 
-import javafx.util.Pair
-
 import java.util.Random
 import kotlin.math.absoluteValue
 
diff --git a/src/main/kotlin/RockPaperSmash/Driver.kt b/src/main/kotlin/RockPaperSmash/Driver.kt
index 9416b6d..fa536e9 100644
--- a/src/main/kotlin/RockPaperSmash/Driver.kt
+++ b/src/main/kotlin/RockPaperSmash/Driver.kt
@@ -11,10 +11,26 @@ import RockPaperSmash.CPU_DIFFICULTY.difficult
 import RockPaperSmash.CPU_DIFFICULTY.easy
 import kotlin.math.absoluteValue
 
-/**
- * Main class. Everything is run here.
- */
-object Driver {
+fun main(args: Array<String>) {
+    try {
+        Driver().menu()
+    } catch (e: Exception) {
+        println("Program prematurely killed. Cleaning up before exiting.")
+        e.printStackTrace()
+        Driver().cleanDir()
+    }
+
+    Runtime.getRuntime().addShutdownHook(object : Thread() {
+
+        override fun start() {
+            println("Exiting.")
+            println("Thanks for playing RockPaperSmash!")
+            Driver().cleanDir()
+        }
+    })
+}
+
+class Driver {
 
     /**
      * Positive responses that can be used to answer dialogs.
@@ -131,26 +147,6 @@ object Driver {
             return Battlefield()
         }
 
-    @JvmStatic
-    fun main(args: Array<String>) {
-        try {
-            menu()
-        } catch (e: Exception) {
-            println("Program prematurely killed. Cleaning up before exiting.")
-            e.printStackTrace()
-            cleanDir()
-        }
-
-        Runtime.getRuntime().addShutdownHook(object : Thread() {
-
-            override fun start() {
-                println("Exiting.")
-                println("Thanks for playing RockPaperSmash!")
-                cleanDir()
-            }
-        })
-    }
-
     /**
      * Has one player choose the [Champion] they would like to play.
      *
@@ -805,7 +801,7 @@ object Driver {
     /**
      * Removes temporary battle files from tmpBattleFiles folder.
      */
-    private fun cleanDir() {
+    public fun cleanDir() {
         TMP_DIR.listFiles()?.mapNotNull {
             file ->
             if (!file.delete()) {
```

Reply:
(xml changes are optional)

Reply:
Done